src/commands/org/logout.ts

```diff
@@ -0,0 +1,215 @@
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import {
+  AuthInfo,
+  AuthRemover,
+  ConfigAggregator,
+  Global,
+  Messages,
+  Mode,
+  OrgAuthorization,
+  OrgConfigProperties,
+} from '@salesforce/core';
+import { Flags, loglevel } from '@salesforce/sf-plugins-core';
+import { Interfaces } from '@oclif/core';
+import { Separator } from 'inquirer';
+import * as chalk from 'chalk';
+import { AuthBaseCommand } from '../../authBaseCommand';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-auth', 'logout');
+const commonMessages = Messages.loadMessages('@salesforce/plugin-auth', 'messages');
+type choice = { name: string; value: OrgAuthorization };
+
+export type AuthLogoutResults = string[];
+
+export default class Logout extends AuthBaseCommand<AuthLogoutResults> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+
+  public static readonly deprecateAliases = true;
+  public static aliases = ['force:auth:logout', 'auth:logout'];
+
+  public static readonly flags = {
+    'target-org': Flags.string({
+      summary: messages.getMessage('flags.target-org.summary'),
+      char: 'o',
+      aliases: ['targetusername', 'u'],
+      deprecateAliases: true,
+    }),
+    all: Flags.boolean({
+      char: 'a',
+      summary: messages.getMessage('all'),
+      description: messages.getMessage('allLong'),
+      required: false,
+      default: false,
+      exclusive: ['target-org'],
+    }),
+    'no-prompt': Flags.boolean({
+      char: 'p',
+      summary: commonMessages.getMessage('noPrompt'),
+      required: false,
+      deprecateAliases: true,
+      aliases: ['noprompt'],
+    }),
+    loglevel,
+  };
+
+  private flags: Interfaces.InferredFlags<typeof Logout.flags>;
+
+  private static buildChoices(orgAuths: OrgAuthorization[], all: boolean): Array<choice | Separator> {
+    const maxUsernameLength = Math.max('Username'.length, ...orgAuths.map((orgAuth) => orgAuth.username.length));
+    const maxAliasLength = Math.max(
+      'Aliases'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.aliases ? orgAuth.aliases.join(',') : '').length)
+    );
+    const maxConfigLength = Math.max(
+      'Configs'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.configs ? orgAuth.configs.join(',') : '').length)
+    );
+    const maxTypeLength = Math.max(
+      'Type'.length,
+      ...orgAuths.map((orgAuth) => {
+        if (orgAuth.isScratchOrg) {
+          return 'Scratch'.length;
+        }
+        if (orgAuth.isDevHub) {
+          return 'DevHub'.length;
+        }
+        if (orgAuth.isSandbox) {
+          return 'Sandbox'.length;
+        }
```

Comment:
Should this also include org shapes?

Reply:
Right now we are not attributing OrgAuthentication with orgs that are 'shapes' or 'snapshots'